### PR TITLE
Add ADR: new application for UI (content management)

### PR DESCRIPTION
**Pull request by pmanrubia**
[Trello card](https://trello.com/c/YhUUAhNU/516-adr-split-ui-into-its-own-app)

### CODE REVIEW

File: doc/arch/adr-011-new-app-for-ui-management.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+ - It seems more consistent with the way GOV.UK has designed other systems.
+
+### Build the UI in the same application (Content Performance Manager) 
+
+Pros:
+
+ - There is no extra infrastructure maintenance / configuration. 
+ - The application is deployed once, there is no need to synchronise the deployments of the backend / frontend.
+ - We can evolve the application without touching the API, which impacts in having a stable API free form frequent versioning.
+ - The development pace is a bit faster because the distance to the source of the data is sorter, and don't depend on remote calls.
```
Comment (chao-xian) [inline doc/arch/adr-011-new-app-for-ui-management.md]: Typo: `shorter` instead of `sorter`

File: doc/arch/adr-011-new-app-for-ui-management.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+ - The architecture is easier to explain.
+ - It seems more consistent with the way GOV.UK has designed other systems.
+
+### Build the UI in the same application (Content Performance Manager) 
+
+Pros:
+
+ - There is no extra infrastructure maintenance / configuration. 
+ - The application is deployed once, there is no need to synchronise the deployments of the backend / frontend.
+ - We can evolve the application without touching the API, which impacts in having a stable API free form frequent versioning.
```
Comment (chao-xian) [inline doc/arch/adr-011-new-app-for-ui-management.md]: Typo: `from` instead of `form`

Comment (pmanrubia) [reply doc/arch/adr-011-new-app-for-ui-management.md]: Fixed in https://github.com/alphagov/content-performance-manager/pull/798/commits/fb7e6237526b0e9e931e77afbac938c45007d2b2

File: doc/arch/adr-011-new-app-for-ui-management.md (hunk 1 above)
Comment (pmanrubia) [reply doc/arch/adr-011-new-app-for-ui-management.md]: Fixed in https://github.com/alphagov/content-performance-manager/pull/798/commits/ed27fb7e337d11ea9f5c3a53ec1c726e66d90297

**Comment by kevindew** [comment]
Hey :wave: 

I'm commenting since you tagged me on Slack. I'm a bit confused about the scope of this ADR though given the discussion yesterday - which probably reveals some of my ignorance overall.

Are we asking here should we build a new application that consumes the data warehouse API or that the new application should be an API and a UI?

I think what would be useful to know here is that if we are building a new application are we killing an old one (content performance manager) and if so what likely timespan is that? As one of the biggest problem I think we face on GOV.UK is we add new applications without killing old ones and then our support burden grows.

File: doc/arch/adr-011-new-app-for-ui-management.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
+# ADR 011: Create a new application for UI management 
+
+18-07-2018
+
+## Context
+
+We are building a UI to provide Content Designers and Managing Editors with
```
[the remark this answers is not in the document]
Comment (pmanrubia) [reply doc/arch/adr-011-new-app-for-ui-management.md]: I have written the information that the team has as of now in 8715b8f80138cbcdc44707646ae78d9ce58f88e1. Unfortunately we don't have all the details yet.

File: doc/arch/adr-011-new-app-for-ui-management.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+### Create a new application to host this functionality
+
+Pros:
+ 
+ - It better reveals the purpose of the Data Warehouse: a source of data for other applications to use
+ - It enables having a robust API so other applications could quickly benefit from having a stable API with well-defined endpoints.
+ - The architecture is easier to explain.
+ - It seems more consistent with the way GOV.UK has designed other systems.
+
```
[the remark this answers is not in the document]
Comment (pmanrubia) [reply doc/arch/adr-011-new-app-for-ui-management.md]: Addressed in c5c4e7d0c0f360d08b160f91fb8f3abf3bfa4a80

File: doc/arch/adr-011-new-app-for-ui-management.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+ - There is no extra infrastructure maintenance / configuration. 
+ - The application is deployed once, there is no need to synchronise the deployments of the backend / frontend.
+ - We can evolve the application without touching the API, which impacts in having a stable API free from frequent versioning.
+ - The development pace is a bit faster because the distance to the source of the data is shorter, and don't depend on remote calls.
+
+## Decision
+
+Create a new application to host the UI for Content Management at scale
+
```
[the remark this answers is not in the document]
Comment (pmanrubia) [reply doc/arch/adr-011-new-app-for-ui-management.md]: I have tried with 4e9ecba158ac2e785aa21d826e58c49a9766f7e3

File: doc/arch/adr-011-new-app-for-ui-management.md [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+### Create a new application to host this functionality
+
+Pros:
+ 
+ - It better reveals the purpose of the Data Warehouse: a source of data for other applications to use
+ - It enables having a robust API so other applications could quickly benefit from having a stable API with well-defined endpoints.
+ - The architecture is easier to explain.
+ - It seems more consistent with the way GOV.UK has designed other systems.
+
+### Build the UI in the same application (Content Performance Manager)
```
[the remark this answers is not in the document]
Comment (pmanrubia) [reply doc/arch/adr-011-new-app-for-ui-management.md]: I have tried to explain it a bit better in 3e8be7bb2cce2d523e44b9cb4edfd606d716208c, but not sure if I have done a good work.

File: doc/arch/adr-011-new-app-for-ui-management.md [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+around content.
+
+Now we have to make a decision about where this new functionality should live. 
+We are exploring two options:
+
+1. Create a new application to host this functionality
+2. Build the UI within the same application (Content Performance Manager)
+
+
+### Create a new application to host this functionality
```
[the remark this answers is not in the document]
Comment (pmanrubia) [reply doc/arch/adr-011-new-app-for-ui-management.md]: I have tried to give more information in 8715b8f80138cbcdc44707646ae78d9ce58f88e1. I have also referenced your comment in the PR as they seem like good ideas to keep the API as simple as possible. Thank you for writing them down.

**Comment by pmanrubia** [comment]
Thank you for your feedback @kevindew, sorry if the ADR wasn't clear enough.

I have tried to answer your questions in 8715b8f80138cbcdc44707646ae78d9ce58f88e1

> I'm commenting since you tagged me on Slack. I'm a bit confused about the scope of this ADR though given the discussion yesterday - which probably reveals some of my ignorance overall.

I tagged you because your feedback is always welcome, but feel free to skip it you are too busy.

> Are we asking here should we build a new application that consumes the data warehouse API or that the new application should be an API and a UI?

This ADR is about having a new UI app that consumes the data from the DW via an API.

> I think what would be useful to know here is that if we are building a new application are we killing an old one (content performance manager) and if so what likely timespan is that? As one of the biggest problem I think we face on GOV.UK is we add new applications without killing old ones and then our support burden grows.

We would not be retiring any application, just adding a new one.

Hopefully this is a bit more clear now.

**Comment by kevindew** [comment]
| Hopefully this is a bit more clear now.

Thanks @pmanrubia your explanation and your changes to this file have made this far clearer. This makes a lot more sense to me now.

Sounds good - btw big +1 to renaming it in line with the naming conventions. I had no idea this was mostly an API

**Comment by pmanrubia** [comment]
I have squashed, rebased and force pushed in preparation for merging

**Pull request merged**